tree/ntuple/v7/inc/ROOT/RField/RFieldFundamental.hxx

```diff
@@ -374,11 +378,20 @@ public:
 
    void AcceptVisitor(Detail::RFieldVisitor &visitor) const final;
 
+   /// Sets this field to use a half precision representation, occupying half as much storage space (16 bits) on disk.
+   /// This is mutually exclusive with `SetTruncated` and `SetQuantized`.
    void SetHalfPrecision();
    /// Set the precision of this field to `nBits`. The remaining (32 - `nBits`) bits will be truncated
    /// from the number's mantissa. `nBits` must be $10 <= nBits <= 31$ (this means that at least 1 bit
    /// of mantissa is always preserved). Note that this effectively rounds the number towards 0.
+   /// This is mutually exclusive with `SetHalfPrecision` and `SetQuantized`.
+   /// \note Calling `SetTruncated(16)` effectively makes this field a `bfloat16` on disk.
    void SetTruncated(std::size_t nBits);
+   /// Sets this field to use a quantized integer representation using `nBits` per value.
+   /// This call promises that this field will only contain values contained in `[minValue, maxValue]` inclusive.
+   /// If a value outside this range is assigned to this field, the behavior is undefined.
```

Comment:
Should we define it? I.e. test and throw?

Reply:
I wanted to keep the specification as vague as possible to give us room for implementing it in the most performant way possible. Maybe this is not necessary, but I think we should have a discussion about it before settling for anything other than UB - we can always specify more in the future, but not go back.

Reply:
I think that's fine for the specification. But in our current implementation, should we / will we throw?

Reply:
Added in [6526106](https://github.com/root-project/root/pull/16390/commits/6526106a85f18eef18f72061f2c410670687d585)